code/controllers/subsystems/job.dm

```diff
@@ -403,6 +403,17 @@
 
 	INVOKE_ASYNC(GLOBAL_PROC, .proc/show_location_blurb, H.client, 30)
 
+	if(SSticker.mode != "Extended" && joined_late)
+		var/antag_count = 0
+		for(var/antag_type in SSticker.mode.antag_tags)
+			var/list/cur_antags = get_antags(antag_type)
+			antag_count += length(cur_antags)
+		for(var/antag_type in SSticker.mode.antag_tags)
+			var/datum/antagonist/A = all_antag_types[antag_type]
+			if(A.can_become_antag(H.mind) && (A.role_type in H.client.prefs.be_special_role) && !(A.flags & ANTAG_OVERRIDE_JOB) && antag_count < A.hard_cap_round && antag_count <= (length(player_list) / SSticker.mode.antag_scaling_coeff))
+				A.add_antagonist(H.mind)
```

Comment:
```suggestion
			if(A.can_become_antag(H.mind) && (A.role_type in H.client.prefs.be_special_role) && !(A.flags & ANTAG_OVERRIDE_JOB) && antag_count < A.hard_cap_round && antag_count <= (length(player_list) / SSticker.mode.antag_scaling_coeff))
```

Some of these are already checked in `add_antagonist()`, you can check the return value of that.

Reply:
i removed some of the checks, lemme know if i missed any